jsonpointer.py

```diff
@@ -153,28 +153,53 @@ def to_last(self, doc):
         if not self.parts:
             return doc, None
 
-        for part in self.parts[:-1]:
-            doc = self.walk(doc, part)
-
-        return doc, self.get_part(doc, self.parts[-1])
-
-
-    def resolve(self, doc, default=_nothing):
+        doc = self.resolve(doc, parts=self.parts[:-1])
+        last = self.parts[-1]
+        ptype = type(doc)
+        if ptype == dict:
+            pass
+        elif ptype == list or isinstance(doc, Sequence):
```

Comment:
Oh, I miss this accidentally to. Normally, this should be used `is` operator since types are some sort of singletons and there is no need to invoke regular comparison routines for them. 

The idea with check for list before call isinstance is an optimization for general case where doc is mostly dict or list. So the idea is to use cheaper operations for common cases and fallback to more wide, but costly, type checks: `isinstance(foo, Sequence)` is 100 times slower when direct type comparison because of function call and unwind all the registered types to those abstract Sequence one.